src/App.tsx

```diff
@@ -20,12 +31,17 @@ export const App: React.FC = () => {
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                value={query}
+                onChange={(event) => {
+                  setQuery(event.target.value);
+                  filterByQuery();
```

Comment:
when i input a new letter it shows me results for previous word, not for current

Reply:
you need to filter before each render, not after each input

Reply:
`setQuery` calls a `render` with previous `visibleMovies`